java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/ResultSetUtility.java

```diff
@@ -0,0 +1,1617 @@
+        this.scale = element.getScale();
+        this.nullable = element.isNullable();
+      }
+
+      private String getLabel() {
+        return "col_" + index;
+      }
+
+      private String getName() {
+        return getLabel();
+      }
+
+      private int getType() {
+        return sqlType;
+      }
+
+      private int getPrecision() {
+        return precision;
+      }
+
+      private int getScale() {
+        return scale;
+      }
+
+      private int isNullable() {
+        return nullable;
+      }
+
+      static MockColumnMetaData fromDataElement(MockDataElement element, int i) throws SQLException {
+        return new MockColumnMetaData(i, element);
+      }
+
+    }
+
+  }
+
+  public static class MockRow {
+    private final ArrayList<MockDataElement> dataElements;
+
+    public MockRow(ArrayList<MockDataElement> elements) {
+      this.dataElements = elements;
+    }
+
+    public MockDataElement getDataElementAtIndex(int idx) throws SQLException {
+      if (idx > dataElements.size()) {
+        throw new SQLException("Unable to find data element at position: " + idx);
+      }
+      return dataElements.get(idx);
+    }
+  }
+
+  public static class MockDataElement {
+    private final Object value;
+    private final int sqlType;
+
+    public MockDataElement(String val) {
+      this(val, Types.VARCHAR);
+    }
+
+    public MockDataElement(Object val, int sqlType) {
+      this.value = val;
+      this.sqlType = sqlType;
+    }
+
+    private boolean isNull() {
+      return value == null;
+    }
+
+    private String getValueAsString() {
+      return value.toString();
+    }
+
+    private int getPrecision() throws SQLException {
+      if (this.sqlType == Types.VARCHAR) {
+        return getValueAsString().length();
+      }
+      throw getExceptionToThrow("Unable to determine precision for data type!");
+    }
+
+    private int getScale() throws SQLException {
+      if (this.sqlType == Types.VARCHAR) {
+        return 0;
+      }
+      throw getExceptionToThrow("Unable to determine precision for data type!");
+    }
+
+    private int isNullable() throws SQLException {
+      if (this.sqlType == Types.VARCHAR) {
+        return ResultSetMetaData.columnNullable;
+      }
+      throw getExceptionToThrow("Unable to determine precision for data type!");
+    }
+
+    public BigDecimal getBigDecimal() throws SQLException {
+      try {
+        return new BigDecimal(getValueAsString());
+      } catch (Exception ex) {
+        throw new SQLException(ex);
+      }
+    }
+
+    public String getString() throws SQLException {
+      return getValueAsString();
+    }
+
+    public boolean getBoolean() throws SQLException {
+      try {
+        return Boolean.parseBoolean(getValueAsString());
+      } catch (Exception ex) {
+        throw new SQLException(ex);
+      }
+    }
+
+    public int getInt() throws SQLException {
+      try {
+        return Integer.parseInt(getValueAsString());
```

Comment:
Hmm, why don't we just cast the object we're holding?

Reply:
That's an option, but the thinking behind my approach here was to enable easy type conversion.  By way of example, the following test succeeds with the current code, but will fail with object casting:

```java
  @Test
  public void testMockDataTypes() throws SQLException {
    ResultSetUtility.MockDataElement element = new ResultSetUtility.MockDataElement(1L, Types.NUMERIC);
    assertEquals(1L, element.getLong());
    assertEquals(1, element.getInt());
    assertEquals("1", element.getString());
  }
```

Reply:
Ah, ok, sounds good.